lib/vm/src/instance.rs

```diff
@@ -761,99 +680,98 @@ impl Instance {
     }
 }
 
-/// A handle holding an `Instance` of a WebAssembly module.
-#[derive(Hash, PartialEq, Eq)]
-pub struct InstanceHandle {
-    instance: *mut Instance,
-
-    /// Whether `Self` owns `self.instance`. It's not always the case;
-    /// e.g. when `Self` is built with `Self::from_vmctx`.
-    ///
-    /// This information is necessary to know whether `Self` should
-    /// deallocate `self.instance`.
-    owned_instance: bool,
+#[derive(Debug)]
+#[repr(C)]
+pub struct InstanceAllocator {
+    ptr_to_dealloc: *const Arc<Self>,
+    ptr_layout: Layout,
+    instance: mem::ManuallyDrop<Instance>,
 }
 
-/// # Safety
-/// This is safe because there is no thread-specific logic in `InstanceHandle`.
-/// TODO: this needs extra review
-unsafe impl Send for InstanceHandle {}
+impl InstanceAllocator {
+    #[inline(always)]
```

Comment:
minor nit but probably unnecessary: Rust is theoretically pretty good at inlining, this attribute therefore only says "if Rust ever decides it's a bad idea to inline this for some reason, do it anyways" which maybe we don't want.

Don't let this block anything though, this is the kind of thing we can figure out later and do a code-base wide review on our use of `#[inline]` and `#[inline(always)]`

Reply:
Anyway, the refactoring removed this :-).